verification/spec/connection-handshake/ch.tla

```diff
@@ -0,0 +1,188 @@
+--------------------------------- MODULE CH ---------------------------------
+
+
+VARIABLES
+    parties,            \* The state of each party.
+    inboundDatagrams    \* For each party, a buffer with any incoming datagrams. 
+
+
+Parties == {"A", "B"}
+
+ConnectionStates == { "UNINIT" }
+
+Null == "none"
+
+ConnectionIDs == {"connAtoB", "connBtoA"}
+
+
+ClientIDs == { "clientA", "clientB" }
+
+
+(***************************** ConnectionEnds *****************************
+    Ref.: https://github.com/informalsystems/ibc-rs/pull/55.
+    
+    A set of connection end records. 
+    A connection end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this connection end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+    
+    - connectionID -- a connection identifier
+      Stores the connection identifier of this connection end.
+    
+    - counterpartyConnectionID -- a connection identifier
+      Stores the connection identifier of the counterparty connection end.
+    
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end. 
+      
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnd ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {Null},
+        clientID : ClientIDs \union {Null},
+        counterpartyConnectionID : ConnectionIDs \union {Null},
```

Comment:
We should probably introduce abstraction for clientId, and connectionId (like what is called Counterparty in Go implementation). Just I would suggest not to use that name of course :) In my view this abstraction should be called ConnectionEnd and what is currently called ConnectionEnd should be Connection. Connection is then defined by two connectionEnds (local and remote, or endAtA and endAtB, or something third.